### ISSUE TITLE
[jaav] False positive CheckResultSet with ResultSet.next() in a while loop

### ISSUE THREAD

**Issue by garydgregory** [pmd/pmd#5031]
**Affects PMD Version:** 

7.1.0

**Rule:**

CheckResultSet 

Please provide the rule name and a link to the rule documentation:

https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#checkresultset

**Description:**

From Apache Commons DbUtils:

```
    public <T> List<T> toBeanList(final ResultSet resultSet, final Class<? extends T> type) throws SQLException {
        final List<T> results = new ArrayList<>();
        if (!resultSet.next()) {
            return results;
        }
        final PropertyDescriptor[] props = this.propertyDescriptors(type);
        final ResultSetMetaData rsmd = resultSet.getMetaData();
        final int[] columnToProperty = this.mapColumnsToProperties(rsmd, props);
        do {
            results.add(this.createBean(resultSet, type, props, columnToProperty));
        } while (resultSet.next()); // NOPMD False positive CheckResultSet
        return results;
    }
```

- git clone  https://gitbox.apache.org/repos/asf/commons-dbutils.git
- git checkout 9159f8b84ae4c43c210c045445b08549e2e1f43e
- mvn pmd:check
- cat target/pmd.xml

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

	<suppressedviolation filename="C:\Users\ggregory\git\a\commons-dbutils\src\main\java\org\apache\commons\dbutils\BeanProcessor.java"
	                     suppressiontype="//nopmd"
	                     msg="Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet."
	                     usermsg=" False positive CheckResultSet

**Running PMD through:** *Maven*

```
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: C:\java\apache-maven-3.9.6
Java version: 17.0.11, vendor: Eclipse Adoptium, runtime: C:\Program Files\Eclipse Adoptium\jdk-17.0.11.9-hotspot
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

**Comment by jbisotti** [comment]
I'm seeing the same this with this code...

```java
        boolean resultSetHasNext = resultSet.next();
        while (resultSetHasNext && resultMap.size() < maxResultCap) {
            ...
            resultSetHasNext = resultSet.next();
        }
```

**Comment by UncleOwen** [comment]
@jbisotti I tried to reproduce using PMD 7.23 and the following code:
```
import java.sql.*;
import java.util.*;

class Foo {
    public void toBeanList(final ResultSet resultSet) throws SQLException {
        Map<String, String> resultMap = new HashMap<>();
        int maxResultCap = 10;

        boolean resultSetHasNext = resultSet.next();
        while (resultSetHasNext && resultMap.size() < maxResultCap) {
            resultSetHasNext = resultSet.next();
        }
    }
}
```
I get no warning.

With the code from @garydgregory I **do** get a warning.
